lib/project_types/extension/commands/serve.rb

```diff
@@ -59,6 +59,16 @@ def start_tunnel_if_required(runtime_configuration)
         return runtime_configuration unless specification_handler.establish_tunnel?(@ctx)
         return runtime_configuration unless runtime_configuration.tunnel_requested?
 
+        return tunnel(runtime_configuration) if should_start_tunnel(runtime_configuration)
```

Comment:
I would rename `tunnel` to `start_tunnel` and `should_start_tunnel` to `can_start_tunnel?`